roles/edpm_nova/meta/argument_specs.yml

```diff
@@ -77,3 +77,8 @@ argument_specs:
           - /etc
           - /opt
           options defaults to "ro" if not specified
+      edpm_nova_healthcheck:
+        type: bool
+        default: true
+        description: |
+          Enable container health check injection
```

Comment:
injection of helathcheck scripts is something that was intentionally not supported and something we objected too

before moving forward with this please reach out to the comptue dfg as this is not a pattern we want to allow 
(injection fo executables including scripts  of any kind into the continers)

Reply:
I'm a bit confused now. Last time we were discussing health checks on slack it seemed to me that you were objecting on baking health checks to container images and wanted to rather bind it to a container as volume. Now you're writing the opposite. I can definitely go on and get health checks to container images via tcib.

Reply:
so my orginal object was to having any healthcheck not directly supported by nova nativly, i.e. external helthchecks of any kind that could affect the execution of the container.

to that was an object to backing them into the image or injecting them.

since nova does not have health check that would mean that until we added native support we would not add any. 

i later found the jira time and realsied that last time we spoke we agreed on using pgrep as a MVP check as an interim solution to have something

provided the healthcheck action is set to None so that it only serves as a status but wont cause the continare to be restarted that is ok.

it is generally not safe to restart nova-compute unless you are sure it is not processing any rpcs.

we have a separate issue to track added graceful shutdown support to nova
https://issues.redhat.com/browse/OSPRH-6601

so a lot of the concern about health checks comes form the fact that restarting the container because a health check failed could break ongoing operation like live migration and result in the VM going to error state.

the pgrep query in theory will only fail if we are in a zombie state or similar which is non recoverable so that is why we previously agreed to use "pgrep -r DRST nova-compute"

i guess we can proceed in that direction.

from my perspective an incorrect healthcheck is more dangous then no helthcheck.
"pgrep -r DRST nova-compute" is not incorrect its just a very minimal incremental change vs no health check.

Reply:
Ok, yeah this healthcheck implementation is for monitoring purposes only. We (as cloudops) don't plan to interfere with container runtime.